packages/wdio-cucumber-framework/src/index.ts

```diff
@@ -481,6 +483,39 @@ class CucumberAdapter {
         })
     }
 }
+/**
+ * Publishes a Cucumber report to a specified URL using NDJSON files from a directory.
+ * @async
+ * @param {string} cucumberMessageDir - The directory path that holds Cucumber NDJSON files.
+ * @returns {Promise<void>} - A Promise that resolves when the report is successfully published.
+ * @throws {Error} - Throws an error if there are issues with file reading or the publishing process.
+ */
+const publishCucumberReport = async (cucumberMessageDir: string): Promise<void> => {
+    const url = process.env.CUCUMBER_PUBLISH_REPORT_URL || 'https://messages.cucumber.io/api/reports'
+    const token = process.env.CUCUMBER_PUBLISH_REPORT_TOKEN
+    if (!token) {
+        return
+    }
+
+    const { headers } = await got(url, {
+        headers: {
+            Authorization: `Bearer ${token}`,
+        }
+    })
+
+    const { location } = headers
+
+    const files = (await readdir(path.resolve(cucumberMessageDir))).filter((file) => path.extname(file) === '.ndjson')
```

Comment:
resolve should only be needed if there's more than one argument.

```
The right-most parameter is considered {to}. Other parameters are considered an array of {from}.

Starting from leftmost {from} parameter, resolves {to} to an absolute path.

If {to} isn't already absolute, {from} arguments are prepended in right to left order, until an absolute path is found. If after using all {from} paths still no absolute path is found, the current working directory is used as well. The resulting path is normalized, and trailing slashes are removed unless the path gets resolved to the root directory.
```

Reply:
Is `.ndjson` a typo?

Reply:
Using the suggested "normalize" approach should work well. I'll implement this change now.